python/cudf/cudf/bindings/avro.pyx

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+from .cudf_cpp cimport *
+from .cudf_cpp import *
+from cudf.bindings.avro cimport reader as avro_reader
+from cudf.bindings.avro cimport reader_options as avro_reader_options
+from libc.stdlib cimport free
+from libcpp.vector cimport vector
+from libcpp.memory cimport unique_ptr
+
+from cudf.dataframe.column import Column
+from cudf.dataframe.dataframe import DataFrame
+from cudf.utils import ioutils
+from cudf.bindings.nvtx import nvtx_range_push, nvtx_range_pop
+
+from io import BytesIO
+import errno
+import os
+
+
+cpdef cpp_read_avro(filepath_or_buffer, columns=None, skip_rows=None,
+                    num_rows=None):
+    """
+    Cython function to call into libcudf API, see `read_avro`.
+
+    See Also
+    --------
+    cudf.io.parquet.read_avro
+    """
+
+    # Setup reader options
+    cdef avro_reader_options options = avro_reader_options()
+    for col in columns or []:
+        options.columns.push_back(str(col).encode())
+
+    # Create reader from source
+    cdef unique_ptr[avro_reader] reader
+    cdef const unsigned char[:] buffer = None
```

Comment:
How does Cython convert a Python NoneType here? To nullptr? I think we should be more explicit.

Reply:
The generated CPP appears to initialize the buffer from: __Pyx_PyObject_to_MemoryviewSlice_ds_unsigned_char__const__(Py_None, 0);

Reply:
I think it would be safer to set this `= NULL` instead. Passing PyObjects around here sounds unnecessary.

Reply:
Hmm. Initializing with NULL conflicts with the later use of "if buffer is not None:". I'll let @j-ieong comment here, since that seems to be the same way it is done for all other readers.

Reply:
If other readers are doing it then lets just carry on, we haven't seen anything explode 😄

Reply:
Yeah... I'm not sure how we check for no slices if it's set to NULL. The Cython documentation have a section on setting and checking for None, but not for NULL.